From cab90d9466974df4e8371fd4cc98c39b73f0bd98 Mon Sep 17 00:00:00 2001
Message-Id: <cab90d9466974df4e8371fd4cc98c39b73f0bd98.1376484223.git.minovotn@redhat.com>
In-Reply-To: <ac0ba67dff2049ff8601fe686b870bba1879b99f.1376484223.git.minovotn@redhat.com>
References: <ac0ba67dff2049ff8601fe686b870bba1879b99f.1376484223.git.minovotn@redhat.com>
From: "Michael S. Tsirkin" <mst@redhat.com>
Date: Tue, 13 Aug 2013 18:30:11 +0200
Subject: [PATCH 5/6] hw/misc: don't create pvpanic device by default

RH-Author: Michael S. Tsirkin <mst@redhat.com>
Message-id: <1376418595-5369-2-git-send-email-mst@redhat.com>
Patchwork-id: 53277
O-Subject: [PATCH RHEL6.5 v2 1/2] hw/misc: don't create pvpanic device by default
Bugzilla: 991100
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: Eric Blake <eblake@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>

Revert "pvpanic: create pvpanic by default for machine 1.5"

This reverts commit baa08ba335b8dafaae4fb96952835a048ca21d93.

Upstream status: closest upstream commit:

commit 7f3e341a008c585deed174eaf1f826c88c67948a
Author: Marcel Apfelbaum <marcel.a@redhat.com>
Date:   Sun Aug 11 18:10:42 2013 +0300

    hw/misc: don't create pvpanic device by default

    This patch is based on Hu Tao's:
    http://lists.nongnu.org/archive/html/qemu-devel/2013-08/msg00124.html

    No need to hard-code pvpanic as part of the machine.
    It can be added with "-device pvpanic" from command line (The next patch).
    Anyway, for backport compatibility it is still part of 1.5
    machine.

    Signed-off-by: Marcel Apfelbaum <marcel.a@redhat.com>
    Reviewed-by: Hu Tao <hutao@cn.fujitsu.com>
    Message-id: 1376233843-19410-2-git-send-email-marcel.a@redhat.com
    Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>

We don't need compatibility with 1.5 so we can simply
revert the patch creating it by default.

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=991100
Brew build: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=6162125
Signed-off-by: Michael S. Tsirkin <mst@redhat.com>
---
 hw/pc.c      | 7 -------
 hw/pc.h      | 3 ---
 hw/pvpanic.c | 7 -------
 3 files changed, 17 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/pc.c      | 7 -------
 hw/pc.h      | 3 ---
 hw/pvpanic.c | 7 -------
 3 files changed, 17 deletions(-)

diff --git a/hw/pc.c b/hw/pc.c
index fe576ea..83c2b5d 100644
--- a/hw/pc.c
+++ b/hw/pc.c
@@ -1114,8 +1114,6 @@ static const char *setdef_cpu_model(const char *model_str,
     }
 }
 
-static bool has_pvpanic = true;
-
 /* PC hardware initialisation */
 static void pc_init1(ram_addr_t ram_size,
                      const char *boot_device,
@@ -1444,10 +1442,6 @@ static void pc_init1(ram_addr_t ram_size,
 	extboot_init(info->bdrv, 1);
     }
 
-    if (has_pvpanic) {
-        pvpanic_init();
-    }
-
 #ifdef CONFIG_KVM_DEVICE_ASSIGNMENT
     if (kvm_enabled()) {
         add_assigned_devices(pci_bus, assigned_devices, assigned_devices_index);
@@ -1755,7 +1749,6 @@ static void rhel_common_init(const char *type1_version,
 static void pc_rhel640_compat(void)
 {
     disable_kvm_sep();
-    has_pvpanic = false;
 }
 
 static void pc_rhel630_compat(void)
diff --git a/hw/pc.h b/hw/pc.h
index f8e6651..8392c5b 100644
--- a/hw/pc.h
+++ b/hw/pc.h
@@ -87,9 +87,6 @@ void *vmmouse_init(void *m);
 #define vmmouse_init(m) do { } while (0)
 #endif
 
-/* pvpanic.c */
-int pvpanic_init(void);
-
 /* pckbd.c */
 
 void i8042_init(qemu_irq kbd_irq, qemu_irq mouse_irq, uint32_t io_base);
diff --git a/hw/pvpanic.c b/hw/pvpanic.c
index 821aa23..bfad31c 100644
--- a/hw/pvpanic.c
+++ b/hw/pvpanic.c
@@ -21,7 +21,6 @@
 #include "qemu-log.h"
 
 #include "hw/fw_cfg.h"
-#include "hw/pc.h"
 
 /* The bit of supported pv event */
 #define PVPANIC_F_PANICKED      0
@@ -91,12 +90,6 @@ static int pvpanic_isa_init(ISADevice *dev)
     return 0;
 }
 
-int pvpanic_init(void)
-{
-    isa_create_simple("pvpanic");
-    return 0;
-}
-
 static ISADeviceInfo pvpanic_isa_info = {
     .qdev.name     = "pvpanic",
     .qdev.size     = sizeof(PVPanicState),
-- 
1.7.11.7

